tests/bsim/babblekit/include/babblekit/testcase.h

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "bs_tracing.h"
+#include "bs_types.h"
+#include "bstests.h"
+
+extern enum bst_result_t bst_result;
+
+/*
+ * @brief Initialize the test framework
+ *
+ * Call this at the start of the test entry point.
+ *
+ * @param timeout_us Timeout (in microseconds) until the timeout callback
+ * @param message String to print to console after init
+ *
+ */
+#define TEST_START(timeout_us, message)                                                            \
+	do {                                                                                       \
+		bst_ticker_set_next_tick_absolute((bs_time_t)timeout_us);                          \
+		bst_result = In_progress;                                                          \
+		bs_trace_info_time(1, "Test start: %s (timeout at %" PRItime "us)\n", message,     \
+				   timeout_us);                                                    \
+	} while (0)
+
+/*
+ * @brief Fail the test and exit
+ *
+ * Printf-like function that also terminates the device with an error code.
+ *
+ * @param ... format-string and arguments
+ *
+ */
+#define TEST_FAIL(...)                                                                             \
+	do {                                                                                       \
+		bst_result = Failed;                                                               \
+		bs_trace_error_time_line(__VA_ARGS__);                                             \
+	} while (0)
```

Comment:
w.r.t. adding `\n` for `TEST_ASSERT` I see that Mesh has a similar macro

```

#define FAIL(msg, ...)                                                         \
	do {                                                                   \
		bst_result = Failed;                                           \
		bs_trace_error_time_line(msg "\n", ##__VA_ARGS__);             \
	} while (0)
```

Maybe it's as simple as

```suggestion
#define TEST_FAIL(msg, ...)                                                                             \
	do {                                                                                       \
		bst_result = Failed;                                                               \
		bs_trace_error_time_line(msg "\n", ##__VA_ARGS__);                                             \
	} while (0)
```

Reply:
doesn't work